.github/workflows/build_webapp.yml

```diff
@@ -0,0 +1,28 @@
+name: Nook Configurator CI
+
+on:
+  pull_request:
+    branches: [ master ]
+    paths:
+    - 'webapp/**'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    container:
+      image:  google/dart:latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: |
+        cd webapp
+        pub get
```

Comment:
let's see if this works in practice. I think Dart have updated their tooling in the latest version of dart so we might need to call this as `dart pub get` instead of `pub get` (but maybe they haven't rolled it out completely yet) so it'll work for a while still